rholang/src/main/bnfc/rholang.cf

```diff
@@ -113,7 +113,7 @@ VPtNegDbl. ValPattern ::= "-" Double;
 --VPtStr. ValPattern ::= String ;
 
 -- Variables
-token Var ((letter | '_')(letter | digit | '_' | '\'')*) ;
+token Var ((letter | '_')(letter | digit | '_' | '-' | '\'')*) ;
```

Comment:
We removed the [-] character from rholang identifiers so that n-1 and n - 1 would mean the same thing.

Reply:
@pyrocto The issue is that Lisp style requires the hyphens. In other words, we need to make primitive calls to function names that have hyphens in them. One proposal I have is convert all underscores in Rholang to hyphens in Rosette. How does that sound?

Reply:
We should be working to get rid of all ambient authority.  Rholang should not be allowed to call arbitrary rosette primitives, only those without authority like arithmetic.  Eventually all the authority will be encapsulated in an actor listening on a "system" channel that can be attenuated and delegated.

We have to have stdout for the SDK to be useful at all, so I don't mind exposing the display method on strings for now, but there's no need to enable writing to files in the SDK; stdout can be piped to a file on the command line if they want files.

Reply:
Well actually regardless of ambient authority, won't we need to call into primitives with hyphens from the system contract, which is written in Rholang? In either case, I guess we can continue this discussion elsewhere.

Reply:
The system actor will likely have parts written in RBL.

Reply:
Making an observation here-  that while this discussion is very good, @KentShikama should have been able to read a document outlining these decisions before cutting his PR. When are we going to have a document that serves this purpose?

Reply:
https://github.com/rchain/rchain/blob/dev/rholang/doc/rholangtut-mercury.md

"Top-level contracts all have the same API: they have one argument, the channel for the system process. The `system` process contains the names of all the channels that can cause side-effects on the node. It is a built-in process that listens for messages consisting of a method name and some arguments."

I suppose that it's not completely clear from this that `system` is the *only* way to get a channel on which to cause side effects.  I should add text to that effect.

All this stuff should also be on the website under a heading "Object Capabilities":
Miller's collection of information about ocaps. http://www.erights.org/
Miller's thesis is a book on ocaps.  http://www.erights.org/talks/thesis/
Miller, Yee, and Shapiro address all the arguments against
capabilities. http://srl.cs.jhu.edu/pubs/SRL2003-02.pdf
Stiegler's picture book of patterns.
http://erights.org/talks/efun/SecurityPictureBook.pdf
Duffy wrote an ocaps OS at Microsoft, parts of which made it into
other products.
http://joeduffyblog.com/2015/11/10/objects-as-secure-capabilities/
Samuel's description of how OOP design leads to ocaps security.
https://reperiendi.wordpress.com/2009/04/07/good-object-oriented-coding-practices-object-capability-security/
Eich says Mozilla has used ocaps for years & regrets it whenever they
don't. https://mail.mozilla.org/pipermail/es-discuss/2013-March/029080.html